src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java

```diff
@@ -195,6 +195,41 @@ public void setEt3ResponseDueDate(CaseData caseData) {
         }
     }
 
+    public void setNextListedDate(CaseData caseData) {
+        List<String> dates = new ArrayList<>();
+        String nextListedDate = "";
+
+        if (CollectionUtils.isNotEmpty(caseData.getHearingCollection())) {
+            for (HearingTypeItem hearingTypeItem : caseData.getHearingCollection()) {
+                dates = getListedDates(hearingTypeItem);
+
+            }
+        for (String date : dates) {
+            if (nextListedDate.equals("") && LocalDate.parse(date).isAfter(LocalDate.now())
```

Comment:
would it make more sense to have a List<LocalDate> and parse the date when adding to the list rather than parsing it multiple times in this if statement?

Reply:
and maybe a test to confirm the logic?

Reply:
test added and date is parsed once only